pkl-core/src/test/kotlin/org/pkl/core/LanguageSnippetTestsEngine.kt

```diff
@@ -34,6 +34,17 @@ import org.pkl.core.http.HttpClient
 import org.pkl.core.project.Project
 import org.pkl.core.util.IoUtils
 
+private val excludedOnJvm22Plus = listOf(Regex(".*analyzeInvalidHttpModule.*"))
+
+private fun currentMajorJavaVersion(): Int =
+  System.getProperty("java.version").substringBefore('.').toInt()
+
+private fun exclusionsForThisJvm(): List<Regex> =
+  when {
+    currentMajorJavaVersion() >= 22 -> excludedOnJvm22Plus
+    else -> emptyList()
+  }
```

Comment:
General facility to add exclusions to `LanguageSnippetTestsEngine` based on the active JVM.